### PR TITLE
Incluster eventing

### PR DESCRIPTION
**Pull request by grego952**
**Description**

Changes proposed in this pull request:

- provide exemplary code for the asynchronous connection between functions

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/10845

### CODE REVIEW

File: incluster_eventing/emitter/config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: event-emitter
+namespace: default
+runtime: nodejs14
+source:
+    sourceType: inline
+apiRules:
+    - name: incomming-http-trigger
+      service:
+        host: incomming.inclusterevent.wookiee.shoot.canary.k8s-hana.ondemand.com
```
Comment (kwiatekus) [inline incluster_eventing/emitter/config.yaml:9]:
```suggestion
        host: incomming.{put your runtime domain here}
```

File: incluster_eventing/emitter/package.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "function-brave-artur8",
```
Comment (kwiatekus) [inline incluster_eventing/emitter/package.json:2]:
```suggestion
  "name": "sanitise-fn",
```

File: incluster_eventing/receiver/package.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "function-affectionate-damian7",
```
Comment (kwiatekus) [inline incluster_eventing/receiver/package.json:2]:
```suggestion
  "name": "store-fn",
```

**Review by kwiatekus — CHANGES_REQUESTED** [review]

File: incluster_eventing/emitter/config.yaml (hunk 1 above)
Comment (kwiatekus) [reply incluster_eventing/emitter/config.yaml:9]: @strekm is it possible for api gateway to resolve domain when definind API rules?
Could I just pass subdomain name? ( i.e `incoming`) and api-gateway would resolve the whole URL?

Comment (cnvergence) [reply incluster_eventing/emitter/config.yaml:9]: when passing only short hostname, api gateway will add the default domain name: https://github.com/cnvergence/api-gateway/blob/main/internal/helpers/default_domain.go

File: incluster_eventing/emitter/config.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+name: event-emitter
+namespace: default
+runtime: nodejs14
+source:
+    sourceType: inline
+apiRules:
+    - name: incomming-http-trigger
+      service:
+        host: incomming
```
Comment (majakurcius) [inline incluster_eventing/emitter/config.yaml:9]:
```suggestion
    - name: incoming-http-trigger
      service:
        host: incoming
```
Noticed a typo here.

**Review by kwiatekus — APPROVED** [review]

**Review by kwiatekus — CHANGES_REQUESTED** [review]
Please provide a content for readme file to describe what purpose this example serves

**Review by kwiatekus — APPROVED** [review]

File: incluster_eventing/README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+The second Function is a message receiver. It subscribes to the given event type and stores the payload.
+
+## Prerequisites
+
+* [Kyma CLI](https://github.com/kyma-project/cli)
+* Kyma installed locally or on a cluster
+
+## Installation
+
+You can find all installation steps in our [Set asynchronous communication between Functions](https://github.com/kyma-project/kyma/blob/b783d9e6dffc47c0e3c31923aff62371b0a46779/docs/03-tutorials/00-serverless/svls-11-set-asynchronous-connection-of-functions.md) tutorial.
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply incluster_eventing/README.md:15]: + The link leads to the forked repository, not the `main` branch 🤔

**Review by klaudiagrz — APPROVED** [review]

**Review by kwiatekus — APPROVED** [review]

**Review by skhalash — APPROVED** [review]

**Review by lilitgh — APPROVED** [review]

**Pull request merged**